internal/home/service.go

```diff
@@ -681,3 +687,64 @@ rc_bg=YES
 
 rc_cmd $1
 `
+
+// secureBinary is used before service.Install(). This function protects AdGuardHome from
+// privilege escalation vulnerabilities caused by writable files
+func secureBinary() error {
+	switch runtime.GOOS {
+	case "windows":
+		// TODO: support windows service support securely
+		// Set file owner to admin/system and public permissions read-only
+		return errors.Error("you currently cannot install adguardhome as a service on window")
+	default:
+		return secureBinaryUnix()
+	}
+}
+
+func secureBinaryUnix() error {
```

Comment:
Please document the function (or the exported API once moved to `aghos`).

See https://github.com/AdguardTeam/CodeGuidelines/blob/master/Go/Go.md#li-4e61a581.